src/library_syscall.js

```diff
@@ -175,27 +175,28 @@ var SyscallsLibrary = {
 
     // arguments handling
 
-    varargs: 0,
+    varargs: undefined,
 
-    get: function(varargs) {
+    get: function() {
+#if ASSERTIONS
+      assert(SYSCALLS.varargs != undefined);
+#endif
       SYSCALLS.varargs += 4;
       var ret = {{{ makeGetValue('SYSCALLS.varargs', '-4', 'i32') }}};
 #if SYSCALL_DEBUG
       err('    (raw: "' + ret + '")');
 #endif
       return ret;
     },
-    getStr: function() {
-      var ret = UTF8ToString(SYSCALLS.get());
+    getStr: function(ptr) {
```

Comment:
can probably (later) get rid of `getStr` and call `UTF8ToString` directly, the logging doesn't seem worth the cost.

Reply:
Sure.